api/src/main/java/net/kyori/adventure/pointer/Pointers.java

```diff
@@ -60,6 +63,30 @@
     return new PointersImpl.BuilderImpl();
   }
 
+  /**
+   * Gets all {@code pointer} in this collection which have the given value type.
+   *
+   * @param valueType the type class of the pointer
+   * @param <V> the type of the pointer
+   * @return an unmodifiable set of pointers filtered by the given type
+   * @since 4.10.0
+   */
+  @SuppressWarnings("unchecked")
+  default <V> @NotNull Set<Pointer<V>> pointers(final @NotNull Class<V> valueType) {
+    return Collections.unmodifiableSet(this.pointers().stream()
```

Comment:
We probably don't need to use streams here, we can construct it without these.

Reply:
Why not use streams? Is there a specific reason behind it? I find it a lot more readable.

Reply:
Streams have overheads, here and there, it's not an issue, but, if we're expecting stuff to be accessing it a lot it can create many needlessly hot areas